# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/pkg_patch/pkg_udev.patch
# Copyright (C) 2006 - 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

We need tiny, early-userspace aware binares statically linked with dietlibc by
default. Thus no .so.

  - Rene Rebe <rene@exactcode.de>

--- udev-083/extras/scsi_id/scsi.h.orig	2006-01-27 01:41:07.000000000 +0100
+++ udev-083/extras/scsi_id/scsi.h	2006-02-01 16:44:28.000000000 +0100
@@ -81,9 +81,9 @@
 #define	SCSI_ID_ASCII	2
 
 struct scsi_id_search_values {
-	u_char	id_type;
-	u_char	naa_type;
-	u_char	code_set;
+	unsigned char	id_type;
+	unsigned char	naa_type;
+	unsigned char	code_set;
 };
 
 /*

--- udev-101/extras/volume_id/Makefile.orig	2006-09-30 14:30:00.000000000 +0200
+++ udev-101/extras/volume_id/Makefile	2006-10-11 20:14:42.000000000 +0200
@@ -36,11 +36,7 @@
 
 $(PROG): %: $(HEADERS) %.o lib/libvolume_id.a
 	$(E) "  LD      " $@
-ifeq ($(strip $(VOLUME_ID_STATIC)),true)
 	$(Q) $(LD) $(LDFLAGS) -o $@ $@.o $(LIBUDEV) lib/libvolume_id.a $(LIB_OBJS)
-else
-	$(Q) $(LD) $(LDFLAGS) -o $@ $@.o $(LIBUDEV) -Llib -lvolume_id $(LIB_OBJS)
-endif
 
 # man pages
 %.8: %.xml
--- udev-109/extras/volume_id/lib/Makefile.orig	2007-04-23 16:22:43.000000000 +0000
+++ udev-109/extras/volume_id/lib/Makefile	2007-04-28 12:17:14.000000000 +0000
@@ -58,7 +58,7 @@
 	libvolume_id.h		\
 	util.h
 
-all: libvolume_id.a $(SHLIB) libvolume_id.pc
+all: libvolume_id.a libvolume_id.pc
 .PHONY: all
 .DEFAULT: all
 
@@ -102,10 +102,7 @@
 
 install: all
 	$(INSTALL_DATA) -D libvolume_id.h $(DESTDIR)$(includedir)/libvolume_id.h
-	$(INSTALL_LIB) -D $(SHLIB) $(DESTDIR)$(libdir)/$(SHLIB)
-	mkdir -p $(DESTDIR)$(usrlibdir)/
-	ln -sf $(SHLIB) $(DESTDIR)$(libdir)/libvolume_id.so.$(SHLIB_CUR)
-	ln -sf $(libdir)/$(SHLIB) $(DESTDIR)$(usrlibdir)/libvolume_id.so
+	$(INSTALL_LIB) -D libvolume_id.a $(DESTDIR)$(usrlibdir)/libvolume_id.a
 	$(INSTALL_DATA) -D libvolume_id.pc $(DESTDIR)$(usrlibdir)/pkgconfig/libvolume_id.pc
 .PHONY: install
 
